tools/js-sdk-release-tools/src/utils/generateInputUtils.ts

```diff
@@ -0,0 +1,90 @@
+import path from "path";
+import { SDKType } from "../common/types";
+import { loadTspConfig } from "../common/utils";
+import { RunningEnvironment } from "./runningEnvironment";
+import { exists } from "fs-extra";
+
+async function isManagementPlaneModularClient(specFolder: string, typespecProjectFolder: string[] | string | undefined) {
+    if (!typespecProjectFolder) {
+        return false;
+    }
+    
+    if (Array.isArray(typespecProjectFolder) && (typespecProjectFolder as string[]).length !== 1) {
+        throw new Error(`Unexpected typespecProjectFolder length: ${(typespecProjectFolder as string[]).length} (expect 1)`);
+    }
+
+    const resolvedRelativeTspFolder = Array.isArray(typespecProjectFolder) ? typespecProjectFolder[0] : typespecProjectFolder as string;
+    const tspFolderFromSpecRoot = path.join(specFolder, resolvedRelativeTspFolder);
+    const tspConfigPath = path.join(tspFolderFromSpecRoot, 'tspconfig.yaml');
+    if (!(await exists(tspConfigPath))) {
+        return false;
+    }
+
+    const tspConfig = await loadTspConfig(tspFolderFromSpecRoot);
+    if (tspConfig?.options?.['@azure-tools/typespec-ts']?.['isModularLibrary'] !== true) {
```

Comment:
how about the case where data plane has enabled isModularLibrary?

Reply:
some of the configurations may not be appilicable for data plane.

Reply:
synced offline. Support management plane for now. Created a [issue ](https://github.com/Azure/azure-sdk-tools/issues/8935) to track